include/deal.II/base/utilities.h

```diff
@@ -1334,7 +1334,11 @@ namespace Utilities
     // see if the object is small and copyable via memcpy. if so, use
     // this fast path. otherwise, we have to go through the BOOST
     // serialization machinery
+#ifdef DEAL_II_HAVE_CXX17
+    if constexpr (std::is_trivially_copyable<T>() && sizeof(T) * N < 256)
+#else
     if (std::is_trivially_copyable<T>() && sizeof(T) * N < 256)
+#endif
```

Comment:
I don't think this makes a difference. The condition is a compile-time constant, so whether or not it is marked as `constexpr`, the compiler will remove either the `if` or the `else` branch. The only difference would be if you'd get a syntax error in one of the branches, but you clearly don't. So I don't see a need for this.

Reply:
These kinds of changes should stop compilers from complaining that we call `memcpy` for non-trivially-copyable objects. On the other hand, it seems that there wasn't a problem in the first place (or did GCC11 complain?).

Reply:
I was just reviewing code for the serialization feature. I noticed that we have this `constexpr` conditional in the other `pack` and `unpack` functions, so I've put it here as well. So this change is not motivated by any problem.